driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -5,23 +5,80 @@ use dfusion_core::models;
 
 use chrono::Utc;
 use log::{debug, error};
-use serde::Deserialize;
-use serde_json::{json, Value};
 use std::collections::{HashMap, HashSet};
 use std::fs::File;
-use std::io::BufReader;
+use std::io::{BufReader, BufWriter, Read, Write};
 use std::process::Command;
 use web3::types::H160;
 
 const RESULT_FOLDER: &str = "./results/tmp/";
 
 type PriceMap = HashMap<u16, u128>;
 
+mod solver_output {
+    use serde::Deserialize;
+    use std::collections::HashMap;
+    use std::vec::Vec;
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Order {
+        pub exec_sell_amount: Option<String>,
+        pub exec_buy_amount: Option<String>,
+    }
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Output {
+        pub orders: Vec<Order>,
+        pub prices: HashMap<String, Option<String>>,
+    }
+}
+
+mod solver_input {
+    use serde::Serialize;
+    use std::collections::HashMap;
+    use std::vec::Vec;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Order {
+        #[serde(rename = "accountID")]
+        pub account_id: String,
+        pub sell_token: String,
+        pub buy_token: String,
+        pub sell_amount: String,
+        pub buy_amount: String,
+        #[serde(rename = "ID")]
+        pub id: String,
+    }
+
+    #[derive(Serialize)]
+    pub struct Fee {
+        pub token: String,
+        pub ratio: f64,
+    }
+
+    pub type Accounts = HashMap<String, HashMap<String, String>>;
+
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Input {
+        pub tokens: Vec<String>,
+        pub ref_token: String,
+        pub accounts: Accounts,
+        pub orders: Vec<Order>,
+        // TODO vk: is this needed even though its never set to anything?
+        pub prices_prev: HashMap<String, String>,
+        pub fee: Option<Fee>,
+    }
+}
+
 pub struct LinearOptimisationPriceFinder {
     // default IO methods can be replaced for unit testing
-    write_input: fn(&str, &Value) -> std::io::Result<()>,
+    write_input: fn(&str, &str) -> std::io::Result<()>,
```

Comment:
nit: would it make sense to use the strict types here and in read_output or does this make the unit tests harder?

Reply:
These functions are not really needed at the moment. The only time we use this is to check whether the `Fee` struct is correctly serialized (in which case `str` makes sense) and we test the individual `serialize_...` functions already.
I'll keep it like this for now and we can change that later if we want to.